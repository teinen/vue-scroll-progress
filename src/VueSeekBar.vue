<template>
  <div id="vue-seek-bar" :style="vueSeekBarStyle"></div>
</template>

<script>
const props = {
  color: {
    type: String,
    required: true
  },
  opacity: {
    type: String,
    required: true
  },
  height: {
    type: String,
    required: true
  }
};

const data = function() {
  return {
    vueSeekBarStyle: {
      width: "1%",
      "background-color": this.color,
      opacity: this.opacity,
      height: `${this.height}px`
    }
  };
};

const methods = {
  scrollHandler() {
    // Get scrolled position
    const scrolledPer = (window.scrollY / this.calcScreenMaxHeight()) * 100;

    if (scrolledPer <= 1) {
      this.vueSeekBarStyle.width = "1%";
    } else {
      this.vueSeekBarStyle.width = scrolledPer.toString() + "%";
    }
  },
  calcScreenMaxHeight() {
    const docEl = document.documentElement;
    return docEl.scrollHeight - docEl.clientHeight;
  }
};

const component = {
  props,
  data,
  methods,
  mounted() {
    // add event listner
    window.addEventListener("scroll", this.scrollHandler);
  },
  destroyed() {
    // remove event listner
    window.removeEventListener("scroll", this.scrollHandler, false);
  }
};

export default component;
</script>

<style lang="scss">
#vue-seek-bar {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}
</style>
