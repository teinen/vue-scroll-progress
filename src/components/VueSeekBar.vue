<template>
  <div id="seek-bar" :style="seekbarStyle"></div>
</template>

<script>
export default {
  props: {
    color: {
      type: String,
      required: true
    },
    opacity: {
      type: String,
      required: true
    },
    height: {
      type: String,
      required: true
    }
  },
  data() {
    return {
      seekbarStyle: {
        width: "1%",
        "background-color": this.color,
        opacity: this.opacity,
        height: `${this.height}px`
      }
    };
  },
  methods: {
    scrollHandler() {
      // Get scrolled position
      const scrolledPer = (window.scrollY / this.calcScreenMaxHeight()) * 100;

      if (scrolledPer <= 1) {
        this.seekbarStyle.width = "1%";
      } else {
        this.seekbarStyle.width = scrolledPer.toString() + "%";
      }
    },
    calcScreenMaxHeight() {
      const docEl = document.documentElement;
      return docEl.scrollHeight - docEl.clientHeight;
    }
  },
  mounted() {
    // add event listner
    window.addEventListener("scroll", this.scrollHandler);
  },
  destroyed() {
    // remove event listner
    window.removeEventListener("scroll", this.scrollHandler, false);
  }
};
</script>

<style lang="scss" scoped>
#seek-bar {
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
}
</style>
