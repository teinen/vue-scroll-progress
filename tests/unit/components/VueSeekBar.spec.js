import { shallowMount } from '@vue/test-utils'
import VueSeekBar from '@/components/VueSeekBar.vue'

const mockProps = {
  color: '#42b983',
  opacity: '0.8',
  height: '5',
}

describe('VueSeekBar.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(VueSeekBar, { props: mockProps })
    expect(wrapper).toMatchSnapshot()
  })
})
