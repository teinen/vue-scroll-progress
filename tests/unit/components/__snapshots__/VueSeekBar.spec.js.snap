// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VueSeekBar.vue renders correctly 1`] = `<div id="vue-seek-bar" style="width: 1%;"></div>`;
